org.eclipse.dartboard/src/org/eclipse/dartboard/launch/LaunchShortcut.java

```diff
@@ -38,57 +44,80 @@
  * @author Jonas Hungershausen
  *
  */
+@SuppressWarnings("restriction")
 public class LaunchShortcut implements ILaunchShortcut {
 
 	@Override
 	public void launch(ISelection selection, String mode) {
+		// Get the active project from the
 		IProject selected = null;
 		if (selection instanceof StructuredSelection) {
 			Object firstElement = ((StructuredSelection) selection).getFirstElement();
-			if (firstElement instanceof IProject) {
-				selected = (IProject) firstElement;
+			if (firstElement instanceof IResource) {
+				selected = ((IResource) firstElement).getProject();
 			}
 		}
 
+		// Shouldn't happen
+		if (selected == null) {
+			MessageDialog.openError(null, Messages.Launch_ConfigurationRequired_Title,
+					Messages.Launch_ConfigurationRequired_Body);
+			return;
+		}
+
+		launchProject(selected, mode);
+	}
+
+	@Override
+	public void launch(IEditorPart editor, String mode) {
+		IResource resource = editor.getEditorInput().getAdapter(IResource.class);
```

Comment:
AFAIK, Here also we can get IProject. editor.getEditorInput().getAdapter(IProject.class)
Please check

Reply:
That doesn't seem to work. The IProject is null here.